### ISSUE TITLE
[Doc]: Connect-AzAccount Parameter Sets Cannot Describe the Authentication Flow Correctly

### ISSUE THREAD

**Issue by msJinLei** [Azure/azure-powershell#25103]
### Type of issue

Other (describe below)

### Feedback

Currently the parameter sets of Connect-AzAccount don't describe the usages and functions

For example
- In the default parameter set "UserWithSubscriptionId (Default)", `subscription` is optional and so why it is named as "WithSubscriptionId"? Further, `UserWithCredential` is also a user flow and with `subscription` .
- For the parameter sets `ServicePrincipalWithSubscriptionId`, `ServicePrincipalCertificateWithSubscriptionId`, `ServicePrincipalCertificateFileWithSubscriptionId` and `AccessTokenWithSubscriptionId`, what's the meaning for "WithSubscriptionId" as the "subscripiton" is optional for all the parameter sets?

### Content source URL

https://github.com/Azure/azure-powershell/blob/main/src/Accounts/Accounts/help/Connect-AzAccount.md

**Comment by isra-fel** [comment]
Name of parameter sets is not exposed to cx. This is not urgent.

**Comment by msJinLei** [comment]
> Name of parameter sets is not exposed to cx. This is not urgent.

@isra-fel I propose to separate device code flow from "UserWithSubscriptionId" and merge "ServicePrincipalCertificateFileWithSubscriptionId" and "ServicePrincipalCertificateWithSubscriptionId"

**Comment by isra-fel** [comment]
Yes it probably makes sense to seperate the parameter sets according to auth flows, as it's more intuitive for users and developers - but we need to be careful not to cause breaking changes.
